be/src/vec/data_types/serde/data_type_date64_serde.cpp

```diff
@@ -235,6 +242,23 @@ void DataTypeDate64SerDe::read_column_from_arrow(IColumn& column, const arrow::A
             v.cast_to_date();
             col_data.emplace_back(binary_cast<VecDateTimeValue, Int64>(v));
         }
+    } else if (arrow_array->type()->id() == arrow::Type::STRING) {
+        // to be compatible with old version, we use string type for date.
+        auto concrete_array = dynamic_cast<const arrow::StringArray*>(arrow_array);
```

Comment:
why modify this code?

Reply:
this is just fix for write_to_arrow which will make beut core

Reply:
seperate the pr

Reply:
relative pr: https://github.com/apache/doris/pull/48053